### PR TITLE
Calendar dropdown and Database Instance Twig UI

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

**Review by orthagh — APPROVED** [review]
Seems good apart minor point: 
Like #16361, massive action button above the datatable should be spaced a little

**Comment by cconard96** [comment]
massive action button ui fixed in other PR.

**Review by orthagh — APPROVED** [review]

**Review by trasher — DISMISSED** [review]
Please rebase to solve conflict

File: src/CalendarSegment.php [hunk 1]
```diff
@@ -388,105 +379,84 @@ public static function showForCalendar(Calendar $calendar)
...
+                'item' => new self(),
+                'calendars_id' => $ID,
+                'days_of_week' => $daysofweek,
+                'begin' => date('H') . ":00",
+                'end' => ((int) date('H') + 1) . ":00",
+                'params' => [
+                    'canedit' => true,
+                ]
+            ];
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
```
Comment (cedric-anne) [inline src/CalendarSegment.php:395]: Please use a dedicated file that will:
1. render this form,
2. include `components/datatable.html.twig` to render the items list.

File: src/Calendar_Holiday.php [hunk 2]
```diff
@@ -109,95 +107,97 @@ public static function showForCalendar(Calendar $calendar)
...
+                'calendars_id' => $ID,
+                'used' => $used,
+                'params' => [
+                    'canedit' => true,
+                    'target' => self::getFormURL()
+                ]
             ];
-            Html::showMassiveActions($massiveactionparams);
+            // The following template doesn't use generic_show_form because of issues with the CommonDBRelation create permission check (canEdit)
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
```
Comment (cedric-anne) [inline src/Calendar_Holiday.php:122]: Same remark here.

File: templates/generic_show_form.html.twig [hunk 3]
```diff
@@ -58,6 +58,8 @@
             <div class="row flex-row align-items-start flex-grow-1">
                <div class="row flex-row">
                   {% block form_fields %}
+                     {% block pre_form_fields %}
+                     {% endblock %}
```
Comment (cedric-anne) [inline templates/generic_show_form.html.twig:62]: This is not necessary.

You could use something like this:
```
{% block form_fields %}
   {# Add your pre_form_fields content here #}

   {{ parent() }}
{% endblock %}
```

```suggestion
```

File: templates/generic_show_form.html.twig [hunk 4]
```diff
@@ -519,7 +521,7 @@
                         ) }}
                      {% endif %}
 
-                     {% if item.isField('size') %}
+                     {% if item.isField('size') and item_type != 'DatabaseInstance' %}
```
Comment (cedric-anne) [inline templates/generic_show_form.html.twig:524]: IMHO, having this kind of specific code here is a bad idea. I think `size` is not a generic field, so it should probably be removed from here and moved in dedicated `Item_DeviceMemory`, `Monitor` and `Database` forms. These are the only GLPI core items that have this field.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: templates/generic_show_form.html.twig (hunk 4 above)
Comment (cconard96) [reply templates/generic_show_form.html.twig:524]: On the other hand, what is this field used for in DatabaseInstance? The `prepareInputForAdd` forcibly removes the field from the input, the form doesn't have an input for it, there is no search option, and the inventory feature code doesn't seem to have anything related to it. Looks like it was maybe just a copy-paste from `Database` and the column could actually be dropped.

Comment (cedric-anne) [reply templates/generic_show_form.html.twig:522]: Maybe it could be dropped, indeed.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**